### PR TITLE
Add year selection page and refactor pervious steps

### PR DESCRIPTION
**Pull request by jrabbott**
Add year selection page and refactor pervious steps

[AB#191585](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/191585)

### CODE REVIEW

File: web/src/EducationBenchmarking.Web/Views/SchoolPlanning/SelectYear.cshtml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+@using EducationBenchmarking.Web.Extensions
+@model EducationBenchmarking.Web.ViewModels.SchoolPlanViewModel
+
```
Comment (J05h-L) [inline web/src/EducationBenchmarking.Web/Views/SchoolPlanning/SelectYear.cshtml:3]: does this page need a title?

File: web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+using EducationBenchmarking.Web.Domain;
+
+namespace EducationBenchmarking.Web.ViewModels;
+
+public class SchoolPlanViewModel
```
Comment (J05h-L) [inline web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanViewModel.cs:5]: mostly for my own learning here - can this be a primary constructor?

File: web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanningViewModel.cs [hunk 3]
```diff
@@ -6,5 +6,7 @@ public class SchoolPlanningViewModel(School school)
 {
```
Comment (J05h-L) [inline web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanningViewModel.cs:6]: Can this model be removed now we have SchoolPlanViewModel?

File: web/src/EducationBenchmarking.Web/Views/SchoolPlanning/SelectYear.cshtml (hunk 1 above)
Comment (jrabbott) [reply web/src/EducationBenchmarking.Web/Views/SchoolPlanning/SelectYear.cshtml:7]: Good spot - pushed change

File: web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanViewModel.cs (hunk 2 above)
Comment (jrabbott) [reply web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanViewModel.cs]: Good spot - pushed change

File: web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanningViewModel.cs (hunk 3 above)
Comment (jrabbott) [reply web/src/EducationBenchmarking.Web/ViewModels/SchoolPlanningViewModel.cs]: Good spot - pushed change

**Review by J05h-L — APPROVED** [review]

**Pull request merged**